compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/FileSystemProjectDirectory.java

```diff
@@ -90,7 +90,10 @@ public Path getPath() {
     private boolean isSpecialDirectory(Path dirName) {
         // TODO Improve this logic in an extensible manner.
         return dirName.toString().equals(ProjectDirConstants.DOT_BALLERINA_DIR_NAME) ||
-                dirName.toString().equals(ProjectDirConstants.TARGET_DIR_NAME);
+                dirName.toString().equals(ProjectDirConstants.TARGET_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.RESOURCE_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.TEST_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.DOT_GIT_DIR_NAME);
```

Comment:
this has unnecessary repetition, also this list can be dynamic in the future. let's have a list.

Reply:
Fixed with c1901bb